app/views/users/show.html.erb

```diff
@@ -24,15 +24,103 @@
     }
     <% end %>
   </style>
-  <div class="user-profile-header" style="<%= user_colors_style(@user) %>" itemscope itemtype="http://schema.org/Person" itemprop="mainEntity">
-    <meta itemprop="url" content="https://dev.to/<%= @user.username %>">
+
+<script type="application/ld+json">
+{
+    "@context": "http://schema.org",
+    "@type": "Person",
+    "mainEntityOfPage": {
+      "@type": "WebPage",
+      "@id": "https://dev.to/<%= @user.username %>"
+    },
+    "url": "<%= user_url(@user) %>",
+    "sameAs": [
+      <% if @user.twitter_username? %>
+        "https://twitter.com/<%= @user.twitter_username %>",
+      <% end %>
+      <% if @user.github_username? %>
+        "https://github.com/<%= @user.github_username %>",
+      <% end %>
+      <% if @user.mastodon_url? %>
+        "<%= @user.mastodon_url %>",
+      <% end %>
+      <% if @user.facebook_url? %>
+        "<%= @user.facebook_url %>",
+      <% end %>
+      <% if @user.linkedin_url? %>
+        "<%= @user.linkedin_url %>",
+      <% end %>
+      <% if @user.behance_url? %>
+        "<%= @user.behance_url %>",
+      <% end %>
+      <% if @user.stackoverflow_url? %>
+        "<%= @user.stackoverflow_url %>",
+      <% end %>
+      <% if @user.dribbble_url? %>
+        "<%= @user.dribbble_url %>",
+      <% end %>
+      <% if @user.medium_url? %>
+        "<%= @user.medium_url %>",
+      <% end %>
+      <% if @user.gitlab_url? %>
+        "<%= @user.gitlab_url %>",
+      <% end %>
+      <% if @user.instagram_url? %>
+        "<%= @user.instagram_url %>",
+      <% end %>
+      <% if @user.twitch_username? %>
+        "<%= @user.twitch_url %>",
+      <% end %>
+      <% if @user.website_url? %>
+        "<%= @user.website_url %>"
+      <% end %>
```

Comment:
this won't work, because if twitch_username is present, but not the website (or any of the combinations before) the array will have a trailing comma, which is invalid in JSON (you're building a JSON document, not a Ruby hash).